lib/screens/home/home_view_model.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:async';
+
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:intl/intl.dart';
+import 'package:survey_flutter/model/survey_model.dart';
+import 'package:survey_flutter/model/profile_model.dart';
+import 'package:survey_flutter/usecases/base/base_use_case.dart';
+
+final homeViewModelProvider =
+    AsyncNotifierProvider.autoDispose<HomeViewModel, void>(HomeViewModel.new);
+
+class HomeViewModel extends AutoDisposeAsyncNotifier<void> {
+  final _surveyCache = <SurveyModel>[];
+
+  final _surveys = StreamController<List<SurveyModel>>();
+  Stream<List<SurveyModel>> get surveys => _surveys.stream;
+
+  final _profile = StreamController<ProfileModel>.broadcast();
+  Stream<ProfileModel> get profile => _profile.stream;
+
+  final _currentDate = StreamController<String>.broadcast();
+  Stream<String> get currentDate => _currentDate.stream;
+
+  final _isError = StreamController<String>();
+  Stream<String> get isError => _isError.stream;
+
+  final _isLoadMore = StreamController<bool>.broadcast();
+  Stream<bool> get isLoadMore => _isLoadMore.stream;
```

Comment:
```suggestion
  final _profile = StreamController<ProfileModel>();
  Stream<ProfileModel> get profile => _profile.stream;

  final _currentDate = StreamController<String>();
  Stream<String> get currentDate => _currentDate.stream;

  final _isError = StreamController<String>();
  Stream<String> get isError => _isError.stream;

  final _isLoadMore = StreamController<bool>();
  Stream<bool> get isLoadMore => _isLoadMore.stream;
```
The Stream variables are being listened once, so we don't need `broadcast`